### PR TITLE
List on notice for DA changes are added

### PR DESCRIPTION
**Pull request by swaroopa86**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [x] the `keep-helm` label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PRL-3347

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by AlokDatta — COMMENTED** [review]
Code coverage should be 100% with the pending file

File: build.gradle [hunk 1]
```diff
@@ -547,9 +547,10 @@ def sonarExclusions = [
   '**uk/gov/hmcts/reform/prl/services/tab/alltabs/AllTabServiceImpl.java',
   '**uk/gov/hmcts/reform/prl/services/cafcass/HearingService.java',
   '**uk/gov/hmcts/reform/prl/services/cafcass/CaseDataService.java',
+  '**uk/gov/hmcts/reform/prl/services/cafcass/Fl401ListOnNoticeController.java',
   '**uk/gov/hmcts/reform/prl/services/ManageOrderService.java',
   '**uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java',
-  '**uk/gov/hmcts/reform/prl/services/dynamicmultiselectlist/DynamicMultiSelectListService.java'
+  '**uk/gov/hmcts/reform/prl/services/dynamicmultiselectlist/DynamicMultiSelectListService.java',
```
[the remark this answers is not in the document]
Comment (swaroopa86) [reply build.gradle:553]: I have not added this in exclusion list i will check the files once and remove if not required in this list

File: build.gradle [hunk 2]
```diff
@@ -547,9 +547,10 @@ def sonarExclusions = [
   '**uk/gov/hmcts/reform/prl/services/tab/alltabs/AllTabServiceImpl.java',
   '**uk/gov/hmcts/reform/prl/services/cafcass/HearingService.java',
   '**uk/gov/hmcts/reform/prl/services/cafcass/CaseDataService.java',
+  '**uk/gov/hmcts/reform/prl/services/cafcass/Fl401ListOnNoticeController.java',
```
[the remark this answers is not in the document]
Comment (swaroopa86) [reply build.gradle:550]: removed the file which is not available

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/fl401listonnotice/Fl401ListOnNoticeController.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                hearingDataService.getHearingData(existingFl401ListOnNoticeHearingDetails,hearingDataPrePopulatedDynamicLists,caseData));
+        } else {
+            caseDataUpdated.put(
+                FL401_LISTONNOTICE_HEARINGDETAILS,
+                ElementUtils.wrapElements(hearingDataService.generateHearingData(hearingDataPrePopulatedDynamicLists,caseData)));
+
+        }
+        List<DynamicListElement> linkedCasesList = hearingDataService.getLinkedCases(authorisation, caseData);
+        caseDataUpdated.put(
+            "linkedCaCasesList",
```
[the remark this answers is not in the document]
Comment (swaroopa86) [reply src/main/java/uk/gov/hmcts/reform/prl/controllers/fl401listonnotice/Fl401ListOnNoticeController.java]: above comments are addressed

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/fl401listonnotice/Fl401ListOnNoticeController.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            hearingDataService.getDynamicList(linkedCasesList));
+        log.info("Linked CA cases List:::: {}",caseDataUpdated.get("linkedCaCasesList"));
+
+        List<DynamicListElement> legalAdviserList = refDataUserService.getLegalAdvisorList();
+        caseDataUpdated.put("legalAdviserList", DynamicList.builder().value(DynamicListElement.EMPTY).listItems(legalAdviserList)
```
[the remark this answers is not in the document]
Comment (swaroopa86) [reply src/main/java/uk/gov/hmcts/reform/prl/controllers/fl401listonnotice/Fl401ListOnNoticeController.java]: above comments are addressed

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/fl401listonnotice/Fl401ListOnNoticeController.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        );
+        Document document = documentGenService.generateSingleDocument(
+            authorisation,
+            caseData,
+            DA_LIST_ON_NOTICE_FL404B_DOCUMENT,
+            false
+        );
+        caseDataUpdated.put("fl401ListOnNoticeDocument", document);
+
+        return AboutToStartOrSubmitCallbackResponse.builder().data(caseDataUpdated).build();
```
[the remark this answers is not in the document]
Comment (swaroopa86) [reply src/main/java/uk/gov/hmcts/reform/prl/controllers/fl401listonnotice/Fl401ListOnNoticeController.java]: above comments are addressed

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/fl401listonnotice/Fl401ListOnNoticeController.java [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+            callbackRequest.getCaseDetails().getData(),
+            CaseData.class
+        );
+        AllocatedJudge allocatedJudge = allocatedJudgeService.getAllocatedJudgeDetails(caseDataUpdated,
+                                                                                       caseData.getLegalAdviserList(), refDataUserService);
+        caseData = caseData.toBuilder().allocatedJudge(allocatedJudge).build();
+        caseDataUpdated.putAll(caseSummaryTabService.updateTab(caseData));
+        caseDataUpdated.put(FL401_LISTONNOTICE_HEARINGDETAILS, hearingDataService
+            .getHearingData(caseData.getFl401ListOnNotice().getFl401ListOnNoticeHearingDetails(),null,caseData));
+        return AboutToStartOrSubmitCallbackResponse.builder().data(caseDataUpdated).build();
```
[the remark this answers is not in the document]
Comment (swaroopa86) [reply src/main/java/uk/gov/hmcts/reform/prl/controllers/fl401listonnotice/Fl401ListOnNoticeController.java]: above comments are addressed

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/fl401listonnotice/Fl401ListOnNoticeController.java [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        caseDataUpdated.put("isFl401CaseCreatedForWithOutNotice", isCaseWithOutNotice);
+        log.info("Check case is created without Notice::====: {}",caseDataUpdated.get("isFl401CaseCreatedForWithOutNotice"));
+
+        if (caseDataUpdated.containsKey(FL401_LISTONNOTICE_HEARINGDETAILS)) {
+            caseDataUpdated.put(
+                FL401_LISTONNOTICE_HEARINGDETAILS,
+                hearingDataService.getHearingData(existingFl401ListOnNoticeHearingDetails,hearingDataPrePopulatedDynamicLists,caseData));
+        } else {
+            caseDataUpdated.put(
+                FL401_LISTONNOTICE_HEARINGDETAILS,
```
[the remark this answers is not in the document]
Comment (swaroopa86) [reply src/main/java/uk/gov/hmcts/reform/prl/controllers/fl401listonnotice/Fl401ListOnNoticeController.java]: above comments are addressed

**Review by AlokDatta — APPROVED** [review]

**Pull request merged**